src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -32,63 +33,93 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
-            case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
+        String message;
+        try {
+            switch (command) {
+                case Command.EAES -> userProgress.setCountryOrigin(CountryOrigin.EAES);
+                case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
+            }
+            userProgressStorage.save(userProgress);
+            message = messagesCreator.getSummaryAnswer(userProgress);
+        } catch (Exception exception) {
+            message = "error";
         }
-        userProgressStorage.save(userProgress);
-        return messagesCreator.getSummaryAnswer(userProgress);
+        return message;
     }
 
     public String processOwnerType(final String command, final String chatID) throws UtilsborException {
         UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
-        switch (command) {
-            case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
-            case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
+        String message;
+        try {
+            switch (command) {
+                case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
+                case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
+            }
+            userProgressStorage.save(userProgress);
+            message = messagesCreator.getSummaryAnswer(userProgress);
+        } catch (StepsQueueException exception) {
+            message = exception.getWrongStep();
         }
-        userProgressStorage.save(userProgress);
-        return messagesCreator.getSummaryAnswer(userProgress);
+        return message;
     }
 
     public String processCarAge(final String command, final String chatID) throws UtilsborException {
         UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
-        switch (command) {
-            case Command.LESS_3_YEARS_AGE -> userProgress.setCarAge(CarAge.LESS_3_YEARS);
-            case Command.BETWEEN_3_AND_7_YEARS_AGE -> userProgress.setCarAge(CarAge.BETWEEN_3_AND_7_YEARS);
-            case Command.MORE_7_YEARS_AGE -> userProgress.setCarAge(CarAge.MORE_7_YEARS);
+        String message;
+        try {
+            switch (command) {
+                case Command.LESS_3_YEARS_AGE -> userProgress.setCarAge(CarAge.LESS_3_YEARS);
+                case Command.BETWEEN_3_AND_7_YEARS_AGE -> userProgress.setCarAge(CarAge.BETWEEN_3_AND_7_YEARS);
+                case Command.MORE_7_YEARS_AGE -> userProgress.setCarAge(CarAge.MORE_7_YEARS);
+            }
+            userProgressStorage.save(userProgress);
+            message = messagesCreator.getSummaryAnswer(userProgress);
+        } catch (StepsQueueException exception) {
+            message = exception.getWrongStep();
```

Comment:
Let's look at it together maybe it could be more suitable to move catch on higher level